Tests/MessagingInApp/MessagingInAppIntegrationTests.swift

```diff
@@ -0,0 +1,161 @@
+@testable import CioMessagingInApp
+import Foundation
+import SharedTests
+import XCTest
+
+class MessagingInAppIntegrationTest: IntegrationTest {
```

Comment:
Great addition !! few tests that i think can us here

- Multiple messages in queue with page rules
-  Test for Navigating Back to Initial Screen After Dismissal

Reply:
Could you elaborate more on the steps in these tests? I'm having a hard time understanding what these have to do with this ticket? I think you're mentioning tests that are good to have that is outside the scope of these changes? 

If you could give scenarios that are in scope, I am happy to write these tests in this PR.

Reply:
The use case here is, if a customer has already sent multiple in-app messages with different page rules, so when we fetch messages we are going to get the list of messages in queue already. We need to verify, if we change route before rendering what message is being displayed then according to page rule. 

The second test, talks about, Being on a page then moving before its rendered and then coming back to the same page.

Reply:
> The second test, talks about, Being on a page then moving before its rendered and then coming back to the same page.

Test `test_givenUserOnScreenDuringFetch_givenUserNavigatedToDifferentScreenWhileMessageLoading_expectShowModalMessageAfterGoBack` should be covering that one. 

> if a customer has already sent multiple in-app messages with different page rules, so when we fetch messages we are going to get the list of messages in queue already. We need to verify, if we change route before rendering what message is being displayed then according to page rule.

I'll add this test and push it up.